pkg/util/kubernetes/apiserver/controllers.go

```diff
@@ -56,16 +63,28 @@ type ControllerContext struct {
 
 // StartControllers runs the enabled Kubernetes controllers for the Datadog Cluster Agent. This is
 // only called once, when we have confirmed we could correctly connect to the API server.
-func StartControllers(ctx ControllerContext) error {
+func StartControllers(ctx ControllerContext) errors.Aggregate {
+	var wg sync.WaitGroup
+	errChan := make(chan error, len(controllerCatalog))
 	for name, cntrlFuncs := range controllerCatalog {
 		if !cntrlFuncs.enabled() {
 			log.Infof("%q is disabled", name)
 			continue
 		}
-		err := cntrlFuncs.start(ctx)
-		if err != nil {
-			log.Errorf("Error starting %q: %s", name, err.Error())
-		}
+
+		// controllers should be started in parallel as their start functions are
+		// blocking until the informers are sync'ed or the sync period timed-out.
+		// for error propagation we rely on a buffered channel to gather errors
+		// from the spawned goroutines.
+		wg.Add(1)
+		go cntrlFuncs.start(ctx, &wg, errChan)
```

Comment:
the fact that adding/removing to the wait group is so far apart in the code seems error prone to me,
WDYT about having a closure here that will have both `wg.Add` & `wg.Done` calling the start synchronously?